bindings/rust/s2n-tls/src/error.rs

```diff
@@ -279,3 +293,61 @@ impl fmt::Display for Error {
 }
 
 impl std::error::Error for Error {}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use errno::set_errno;
+
+    #[test]
+    fn to_io_error() -> Result<(), Box<dyn std::error::Error>> {
+        // This relies on an implementation detail of s2n-tls errors,
+        // and could make this test brittle. However, the alternative
+        // is a real handshake producing a real IO error, so just updating
+        // this test if the definition of an IO error changes might be easier.
+        let s2n_io_error_code = 1 << 26;
+        let s2n_io_error = Error::Code(s2n_io_error_code, Errno(0));
+        assert_eq!(Some(ErrorType::IOError), s2n_io_error.kind());
+
+        // IO error
+        {
```

Comment:
I know this is s2n-tls C convention but it's better to have individual functions for each test. It's especially nice for when you get a failure: it doesn't return early which makes it easier identify the cause. 

It also makes tests go faster since each function is run in a separate thread by default.